csrc/scheduler/registry.cpp

```diff
@@ -90,6 +89,15 @@ bool checkCanSchedule(Fusion* fusion, SchedulerType scheduler_type) {
     return false;
   }
 
+  // check PreprocessGroupedMatmulInputSf's output is in global memory by
+  // forcing it as a fusion segment output
+  if (registry_utils::SchedulerTopologyChecker::hasConsumerOfNonIndexableOps(
+          fusion)) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        scheduler_type, "Fusion has consumer of non indexable ops.");
+    return false;
+  }
```

Comment:
Note for myself. I think I need to do the same for offsets. promoting memory type to shared/global so each thread can access the entire offsets in case they are computed within the kernel.

I'm not sure if sync analysis during lowering would have done that for me automatically.